pkg/http/server.go

```diff
@@ -112,10 +114,19 @@ func (s *server) useProxy(next fasthttp.RequestHandler) fasthttp.RequestHandler
 				ctx.RemoteAddr(),
 				ctx.Host(),
 				proto))
-		// Add X-Forwarded-For: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For
-		ctx.Request.Header.Add("X-Forwarded-For", ctx.RemoteAddr().String())
-		// Add X-Forwarded-Host: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host
-		ctx.Request.Header.Add("X-Forwarded-Host", fmt.Sprintf("%s", ctx.Host()))
+
+		hostAddress, err := utils.GetHostAddress()
+		if err == nil {
+			// Add X-Forwarded-For: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For
+			ctx.Request.Header.Add("X-Forwarded-For", hostAddress)
+		}
+
+		hostName, err := os.Hostname()
+		if err == nil {
+			// Add X-Forwarded-Host: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host
+			ctx.Request.Header.Add("X-Forwarded-Host", hostName)
+		}
+
```

Comment:
We only need these headers for service invocation. I do not think we need this for the other APIs. So we do not need this middleware.

Reply:
These proxy headers is required only when user app uses remote call. so we can move this code to invokeRemote code.

https://github.com/dapr/dapr/blob/955c30b10756b6e92bc2bc300e00b5ef299db94e/pkg/messaging/direct_messaging.go#L119

Or we can add this headers in CallLocal internal gRPC.

Normally, proxy server does this. So adding headers inside CallLocal RPC makes sense to me. then we can support proxy headers for both http and gRPC protocols.

cc/ @yaron2 @amanbha @jjcollinge

Reply:
👍 Yeh good point - CallLocal makes sense

Reply:
To me placing this logic in invokeRemote makes sense but I don't understand how this can be moved to CallLocal.

As per my understanding CallLocal is only invoked on the callee's sidecar, how can we access host name and host address of caller here?

Reply:
both are fine. but I believe there are grpc helper to get the remote address. If it is hard to implement it, populating forward headers in invokeRemote is fine.

Reply:
As per the discussion offline the logic is moved to invokeRemote because it is unknown how to find clients host name on server side - checked both the request context and metadata.